pkg/persistence/beanstalkd.go

```diff
@@ -0,0 +1,110 @@
+package persistence
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"go.uber.org/zap"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/iwanbk/gobeanstalk"
+	"github.com/pkg/errors"
+)
+
+var beanstalkdWriteOnlyError = errors.New("This dummy storage engine (\"beanstalkd\") is write-only")
```

Comment:
Probably we need to make common error for this type of "not implemented" or "magneticod only" errors like I did with `SimpleTorrentSummary` in `interface.go`.

Reply:
we could create a WriteOnly interface that implements all the functions which read on data (GetNumberOfTorrents, get Torrent, Getdata etc.). Then Beanstalk (or any other kind of write-only db like StdOut) could build on it implementing the remaining "write" function necessary to get to a full Database implementation

Reply:
@lruggieri Totally agree with this idea. But I'm not ready to do this refactoring myself.

Reply:
@skobkin I did a PR on your own branch, implementing this kind of interface. See if you like it.

Reply:
@lruggieri I saw it. Looks good *for me*. But it's would be better to propose such changes to the upstream and then if it's approved I can rebase my PR on them.

Proposing both in my PR would not be good for segregation of proposed changes.